src/utils/api/helper.ts

```diff
@@ -262,6 +262,29 @@ const activePaymentTask = (
               let reason;
               if (responseType.status === 200) {
                 reason = "";
+                const cartInfo = getSessionItem(SessionItems.cart) as
+                  | Cart
+                  | undefined;
+                if (cartInfo !== undefined) {
+                  const updatedPaymentNotices = cartInfo.paymentNotices.map(
+                    (paymentNotice) => {
+                      const updatedAmount = responseType.value.payments.find(
+                        (p) =>
```

Comment:
[**suggestion (if-minor)**:](https://conventionalcomments.org/) _map_ + _find_ makes this an $O(n^2)$ algorithm, what if we first map the actualized amounts from the activate response into a `RptId -> amount` map and then update the amounts?

Reply:
You're right! Solved with 3c936a5 😁